src/App.tsx

```diff
@@ -1,14 +1,81 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Loader } from './components/Loader';
+import { SelectFilter } from './types/SelectFilter';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
-import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectFilter, setSelectFilter] = useState<SelectFilter>(
+    SelectFilter.ALL,
+  );
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+
+  const fetchTodos = useCallback(async () => {
+    setIsLoading(true);
+
+    try {
+      const responseTodos = await getTodos();
+
+      setTodos(responseTodos);
+    } catch (error) {
+      throw new Error('Something went wrong...');
```

Comment:
and you will have unhandled error this way, create state for error message, save it there and use conditional render to show it